test/e2e/deploy_reaper_test.go

```diff
@@ -83,16 +70,76 @@ var _ = Describe("Deploy Reaper with Cassandra backend", func() {
 			err = framework.Client.Create(context.Background(), reaper)
 			Expect(err).ToNot(HaveOccurred(), "failed to create reaper object")
 
+			By("wait for cass-operator to be ready")
+			err = framework.WaitForCassOperatorReady(namespace)
+			Expect(err).ToNot(HaveOccurred(), "failed waiting for cass-operator to become ready")
+
+			By("wait for cassdc to be ready")
+			cassdcRetryInterval := 15 * time.Second
+			cassdcTimeout := 7 * time.Minute
+			err = framework.WaitForCassDcReady(cassdcKey, cassdcRetryInterval, cassdcTimeout)
+			Expect(err).ToNot(HaveOccurred(), "failed waiting for cassdc to become ready")
+
 			By("wait for reaper to become ready")
 			reaperKey := types.NamespacedName{Namespace: reaper.Namespace, Name: reaper.Name}
 			err = framework.WaitForReaperReady(reaperKey, 10*time.Second, 3*time.Minute)
 			Expect(err).ToNot(HaveOccurred(), "failed waiting for reaper to become ready")
 
+			// This should be ready at the same time as cassdc
+
 			By("wait for the cluster to get registered with reaper")
 			err = framework.WaitForReaper(reaperKey, 10*time.Second, 3*time.Minute, func(reaper *api.Reaper) bool {
-				return len(reaper.Status.Clusters) == 1 && reaper.Status.Clusters[0] == cassdc.Spec.ClusterName
+				return len(reaper.Status.Clusters) == 1
+			})
+			Expect(err).ToNot(HaveOccurred(), "failing waiting for cluster to get registered")
+
+			// Remove Cluster and see that it's being readded
+			By("removing cluster status and expecting it to reappear")
+			err = framework.Client.Get(context.Background(), reaperKey, reaper)
+			Expect(err).ToNot(HaveOccurred(), "failed to refresh reaper object")
+
+			reaperPatch := client.MergeFrom(reaper.DeepCopy())
+			reaper.Status.Clusters = make([]string, 0)
+			err = framework.Client.Status().Patch(context.Background(), reaper, reaperPatch)
+
+			err = framework.WaitForReaper(reaperKey, 10*time.Second, 3*time.Minute, func(reaper *api.Reaper) bool {
+				return len(reaper.Status.Clusters) == 0
+			})
+			Expect(err).ToNot(HaveOccurred(), "failing waiting for clusters to get removed")
+
+			err = framework.WaitForReaper(reaperKey, 10*time.Second, 3*time.Minute, func(reaper *api.Reaper) bool {
+				return len(reaper.Status.Clusters) == 1
+			})
+			Expect(err).ToNot(HaveOccurred(), "failing waiting for clusters to get re-registered")
+		})
+
+		Specify("More clusters are created", func() {
+			// Since we didn't clean the previous resources, we don't need to redeploy everything
```

Comment:
Are tests guaranteed to execute in the order in which they are declared?

More importantly, this limits our ability to execute tests in parallel in a CI environment. cass-operator puts every test in a separate suite which makes it really easy to run them in parallel. I don't know if we need to do that but I figured separate test files and `Describe` block would facilitate running them in parallel.

Reply:
In this case if I understood the documentation correctly they should:

"Ginkgo’s default behavior is to only permute the order of top-level containers – the specs within those containers continue to run in the order in which they are specified in the test file. "